schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
+}
+
+type AppUnstakeMsg @entity {
+  id: ID!
+  application: Application! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type DelegateToGatewayMsg @entity {
+  id: ID!
+  application: Application! @index
+  gateway: Gateway! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type UndelegateToGatewayMsg @entity {
+  id: ID!
+  application: Application! @index
+  gateway: Gateway! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type TransferApplicationMsg @entity {
+  id: ID!
+  sourceApplication: Application! @index
+  destinationApplication: Application! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type Application @entity {
+  # id is the address
+  id: ID!
+  account: Account!
+  stake: Coin!
+  # 0 = Staked, 1 = Unstaking, 2 = Unstaked
+  status: Int!
+  # Block where the msg to unstake the application was received
+  unstakingStartBlock: Block
+  # Height of the block where the application is going to be unstaked/unbonded. Received with the EventApplicationUnbondingBegin event.
+  unstakingHeight: BigInt
+  # Reason for the application to be unstaked/unbonded. Received with the EventApplicationUnbondingBegin event.
+  # 0 = ELECTIVE, 1 = BELOW_MIN_STAKE, 2 = TRANSFERRED
+  unstakingReason: Int
+  # Block where the application was unstaked/unbonded. Received with the EventApplicationUnbondingEnd event.
+  unstakedAtBlock: Block
+  stakeMsgs: [AppStakeMsg] @derivedFrom(field: "application")
+  unstakeMsgs: [AppUnstakeMsg] @derivedFrom(field: "application")
+  delegateMsgs: [DelegateToGatewayMsg] @derivedFrom(field: "application")
+  undelegateMsgs: [UndelegateToGatewayMsg] @derivedFrom(field: "application")
+  sourceTransferMsgs: [TransferApplicationMsg] @derivedFrom(field: "sourceApplication")
+  destinationTransferMsgs: [TransferApplicationMsg] @derivedFrom(field: "destinationApplication")
+  sourceTransferBeginEvents: [TransferApplicationBeginEvent] @derivedFrom(field: "source")
+  destinationTransferBeginEvents: [TransferApplicationBeginEvent] @derivedFrom(field: "destination")
+  sourceTransferEndEvents: [TransferApplicationEndEvent] @derivedFrom(field: "source")
+  destinationTransferEndEvents: [TransferApplicationEndEvent] @derivedFrom(field: "destination")
+  sourceTransferErrorEvents: [TransferApplicationErrorEvent] @derivedFrom(field: "source")
+  destinationTransferErrorEvents: [TransferApplicationErrorEvent] @derivedFrom(field: "destination")
+  unstakingBeginEvents: [ApplicationUnbondingBeginEvent] @derivedFrom(field: "application")
+  unstakingEndEvents: [ApplicationUnbondingEndEvent] @derivedFrom(field: "application")
+  transferredFromAt: Block
+  transferFrom: Application
+  transferTo: Application
+  transferToEndedAt: Block
+  # Account because we don't know if the destination is an application
+  transferringTo: Account
+  transferEndsAtHeight: BigInt
+}
+
+type ApplicationDelegatedToGateway @entity {
+  id: ID!
+  gateway: Gateway!
+  application: Application!
+}
```

Comment:
If the purpose of this is to provide a many-to-many join table, this won't be necessary because the application data structure [holds foreign key references](https://github.com/pokt-network/poktroll/blob/main/proto/poktroll/application/types.proto#L30) to the gateways to which it is delegated, and as of [#888](https://github.com/pokt-network/poktroll/pull/888), gateways will also [hold foreign key references](https://github.com/pokt-network/poktroll/pull/888/files#diff-29ad6f0bd019654d0f60a67d3ccb937d5fcdff7a8ae939e9b61a52406ee550b8R18) to their delegated applications.

Reply:
We need the many-to-many relationship here because an application can be delegated to many gateways and a gateway can be delegating many applications.